src/tests/java/picard/analysis/directed/CollectTargetedMetricsTest.java

```diff
@@ -0,0 +1,160 @@
+        return CollectTargetedPcrMetrics.class.getSimpleName();
+    }
+
+    //create a samfile with 40000 reads for testing whether a cap is found.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final int numReads = 40000;
+        final String readName = "TESTBARCODE";
+
+        tempSamFile = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        final File tempSamFileUnsorted = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        tempSamFileUnsorted.deleteOnExit();
+        tempSamFile.deleteOnExit();
+
+        BufferedLineReader bufferedLineReader = null;
+        try {
+            bufferedLineReader = new BufferedLineReader(new FileInputStream(dict));
+        } catch (final FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        final SAMTextHeaderCodec codec = new SAMTextHeaderCodec();
+        final SAMFileHeader header = codec.decode(bufferedLineReader, dict.toString());
+        header.setSortOrder(SAMFileHeader.SortOrder.unsorted);
+
+        final String separator = ":";
+        final int ID = 1;
+        final int max = 15000;
+        final int min = 1;
+        final Random rg = new Random(5);
+
+        readGroupRecord.setSample(sample);
+        readGroupRecord.setPlatform(platform);
+        readGroupRecord.setLibrary(library);
+        readGroupRecord.setPlatformUnit(readGroupId);
+        header.addReadGroup(readGroupRecord);
+        setBuilder.setReadGroup(readGroupRecord);
+        setBuilder.setUseNmFlag(true);
+
+        setBuilder.setHeader(header);
+
+        for (int i = 0; i < numReads; i++) {
+            final int start = rg.nextInt(max) + min;
+            final String newReadName = readName + separator + ID + separator + i;
+            setBuilder.addPair(newReadName, 0, start+ID, start+ID+99);
+        }
+
+        final SAMFileWriter writer = new SAMFileWriterFactory()
+                .setCreateIndex(true).makeBAMWriter(header, false, tempSamFileUnsorted);
+
+        for (final SAMRecord record : setBuilder) {
+            writer.addAlignment(record);
+        }
+        writer.close();
+
+        //sort the temp file
+        final SortSam sorter = new SortSam();
+        final String[] args = new String[]{"INPUT=" + tempSamFileUnsorted.getAbsolutePath(), "OUTPUT=" + tempSamFile.getAbsolutePath(), "SORT_ORDER=coordinate"};
+
+        sorter.instanceMain(args);
+    }
+
+    @Test //test all collection levels
+    public void test() throws IOException{
+        runMaxReadsTest(tempSamFile);
+        runEmptyIntervalTest(tempSamFile);
+    }
+
+    public void runMaxReadsTest(final File input) throws IOException {
+        final File outfile = File.createTempFile("test", ".TargetedMetrics_Coverage");
+        final File perTargetOutfile = File.createTempFile("perTarget", ".perTargetCoverage");
+        final String referenceFile = "testdata/picard/quality/chrM.reference.fasta";
```

Comment:
Better extract this to the top of the class, you are also using this in a different functions below.